contributed_definitions/nyaml/NXpeak.yaml

```diff
@@ -1,59 +1,140 @@
+    \@type:
+      doc: |
+        Specification of the definition of `width`.
+      enumeration:
+        fwhm:
+          doc: |
+            Full Width at Half Maximum of the peak.
+        hwhm:
+          doc: |
+            Half Width at Half Maximum of the peak.
+        half_widths_asymmetric:
+          doc: |
+            For asymmetrical peaks, the half-widths of each side of the peak at
+            half maximum intensity is given.
+        other:
+          doc: |
+            For other peak width definitions.
+  functional_form:
     doc: |
-      In the case of an analytical description (or if peak_model is other) this
+      A formula to describe the line shape using set of parameters as entered
+      by the `peak_parameters` fields.
+  peak_parameters(NXparameters):
+    doc: |
+      In the case of an analytical description (or if description is other) this
       collection holds parameter of (and eventually) the functional form.
       For example in the case of Gaussians mu, sigma, cut-off values,
       and background intensity are relevant parameter.
+  total_area(NX_NUMBER):
+    unit: NX_ANY
+    doc: |
+      Total area under the curve (can also be used for the total area minus any
+      background values.)
+  relative_area(NX_NUMBER):
+    unit: NX_ANY
+    doc: |
+      Relative area of this peak compared to other peaks in an NXpeak_model.
+      
+      The relative area can simply be derived by dividing the total_area by the total
+      area of all peaks or by a more complicated method (e.g., by additionally dividing
+      by the relative_sensitivity_factor). Details should be stored in the quantification
+      description of NXpeak_model.
+  relative_sensitivity_factor(NX_NUMBER):
+    unit: NX_UNITLESS
+    doc: |
+      Relative sensitivity for this peak, to be used in quantification in an NXpeak_model.
+      
+      As an example, in X-ray spectroscopy could depend on the energy scale (see position),
+      the ionization cross section, and the element probed.
+  area_constraint(NXparameters):
+    doc: |
+      Constraint on the minimum and maximum area of this peak.
+    minimal_value(NX_NUMBER):
+      unit: NX_ANY
+    maximal_value(NX_NUMBER):
+      unit: NX_ANY
+    depends_on:
+      doc: |
+        Link to a peak the area constraint depends on.
+  width_constraint(NXparameters):
+    doc: |
+      Constraint on the minimum and maximum width of this peak.
+    minimal_value(NX_NUMBER):
+      unit: NX_ANY
+    maximal_value(NX_NUMBER):
+      unit: NX_ANY
+    depends_on:
+      doc: |
+        Link to a peak the width constraint depends on.
+  position_constraint(NXparameters):
+    doc: |
+      Constraint on the minimum and maximum position of this peak on the abscissa.
+    minimal_value(NX_NUMBER):
+      unit: NX_ANY
+    maximal_value(NX_NUMBER):
+      unit: NX_ANY
+    depends_on:
+      doc: |
+        Link to a peak the position constraint depends on.
```

Comment:
Constraints should be possible to store with each fit parameter, however this should be done at the place where the actual fit is defined, I think, and not where the function for a peak is defined (maybe).

Reply:
Now, each peak has a fit_function containing NXfit_single_parameters or NXfit_repeated_parameters. These parameters have min and max values.